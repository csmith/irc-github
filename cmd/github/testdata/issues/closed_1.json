{
  "action": "closed",
  "issue": {
    "url": "https://api.github.com/repos/greboid/irc/issues/9",
    "repository_url": "https://api.github.com/repos/greboid/irc",
    "labels_url": "https://api.github.com/repos/greboid/irc/issues/9/labels{/name}",
    "comments_url": "https://api.github.com/repos/greboid/irc/issues/9/comments",
    "events_url": "https://api.github.com/repos/greboid/irc/issues/9/events",
    "html_url": "https://github.com/greboid/irc/issues/9",
    "id": 590904706,
    "node_id": "MDU6SXNzdWU1OTA5MDQ3MDY=",
    "number": 9,
    "title": "[BUG] Panic when using streaming API",
    "user": {
      "login": "csmith",
      "id": 189133,
      "node_id": "MDQ6VXNlcjE4OTEzMw==",
      "avatar_url": "https://avatars0.githubusercontent.com/u/189133?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/csmith",
      "html_url": "https://github.com/csmith",
      "followers_url": "https://api.github.com/users/csmith/followers",
      "following_url": "https://api.github.com/users/csmith/following{/other_user}",
      "gists_url": "https://api.github.com/users/csmith/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/csmith/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/csmith/subscriptions",
      "organizations_url": "https://api.github.com/users/csmith/orgs",
      "repos_url": "https://api.github.com/users/csmith/repos",
      "events_url": "https://api.github.com/users/csmith/events{/privacy}",
      "received_events_url": "https://api.github.com/users/csmith/received_events",
      "type": "User",
      "site_admin": false
    },
    "labels": [
      {
        "id": 1873189829,
        "node_id": "MDU6TGFiZWwxODczMTg5ODI5",
        "url": "https://api.github.com/repos/greboid/irc/labels/bug",
        "name": "bug",
        "color": "d73a4a",
        "default": true,
        "description": "Something isn't working"
      }
    ],
    "state": "closed",
    "locked": false,
    "assignee": null,
    "assignees": [

    ],
    "milestone": null,
    "comments": 0,
    "created_at": "2020-03-31T08:30:22Z",
    "updated_at": "2020-03-31T10:23:18Z",
    "closed_at": "2020-03-31T10:23:18Z",
    "author_association": "CONTRIBUTOR",
    "body": "**Describe the bug**\r\nSeemingly every time a plugin accesses the streaming API the bot panics\r\n\r\n**To Reproduce**\r\nSteps to reproduce the behavior:\r\n1. Run bot\r\n2. Run streaming API plugin\r\n\r\n**Expected behavior**\r\nNo panics.\r\n\r\n```\r\npanic: assignment to entry in nil map\r\n\r\ngoroutine 45 [running]:\r\ngithub.com/greboid/irc/irc.(*EventManager).SubscribeChannelPart(0xc0001653f0, 0xc0004fe500)\r\n\tgithub.com/greboid/irc/irc/messageBus.go:52 +0xbb\r\ngithub.com/greboid/irc/rpc.(*pluginServer).GetMessages(0xc0001653e0, 0xc0000e2690, 0x9e7640, 0xc000388760, 0x0, 0x0)\r\n\tgithub.com/greboid/irc/rpc/server.go:97 +0x186\r\ngithub.com/greboid/irc/rpc._IRCPlugin_GetMessages_Handler(0x8ba3c0, 0xc0001653e0, 0x9e5de0, 0xc0004fe4c0, 0x0, 0x0)\r\n\tgithub.com/greboid/irc/rpc/plugin.pb.go:435 +0x109\r\ngithub.com/grpc-ecosystem/go-grpc-middleware/auth.StreamServerInterceptor.func1(0x8ba3c0, 0xc0001653e0, 0x9e5f60, 0xc0002a00c0, 0xc0004fe460, 0x9516a0, 0xc0004fe480, 0xc0004fe460)\r\n\tgithub.com/grpc-ecosystem/go-grpc-middleware@v1.2.0/auth/auth.go:66 +0x148\r\ngithub.com/grpc-ecosystem/go-grpc-middleware.ChainStreamServer.func1.1.1(0x8ba3c0, 0xc0001653e0, 0x9e5f60, 0xc0002a00c0, 0xc00062bc58, 0x40c9c8)\r\n\tgithub.com/grpc-ecosystem/go-grpc-middleware@v1.2.0/chain.go:49 +0x5f\r\ngithub.com/grpc-ecosystem/go-grpc-middleware.ChainStreamServer.func1(0x8ba3c0, 0xc0001653e0, 0x9e5f60, 0xc0002a00c0, 0xc0004fe460, 0x9516a0, 0x9e4060, 0xc0000e2660)\r\n\tgithub.com/grpc-ecosystem/go-grpc-middleware@v1.2.0/chain.go:58 +0xcf\r\ngoogle.golang.org/grpc.(*Server).processStreamingRPC(0xc00007e000, 0x9e8420, 0xc000182480, 0xc00029c100, 0xc000165470, 0xd07780, 0x0, 0x0, 0x0)\r\n\tgoogle.golang.org/grpc@v1.28.0/server.go:1336 +0x511\r\ngoogle.golang.org/grpc.(*Server).handleStream(0xc00007e000, 0x9e8420, 0xc000182480, 0xc00029c100, 0x0)\r\n\tgoogle.golang.org/grpc@v1.28.0/server.go:1409 +0xc62\r\ngoogle.golang.org/grpc.(*Server).serveStreams.func1.1(0xc0004984c4, 0xc00007e000, 0x9e8420, 0xc000182480, 0xc00029c100)\r\n\tgoogle.golang.org/grpc@v1.28.0/server.go:746 +0xa1\r\ncreated by google.golang.org/grpc.(*Server).serveStreams.func1\r\n\tgoogle.golang.org/grpc@v1.28.0/server.go:744 +0xa1\r\n```"
  },
  "repository": {
    "id": 243344149,
    "node_id": "MDEwOlJlcG9zaXRvcnkyNDMzNDQxNDk=",
    "name": "irc",
    "full_name": "greboid/irc",
    "private": false,
    "owner": {
      "login": "greboid",
      "id": 189618,
      "node_id": "MDQ6VXNlcjE4OTYxOA==",
      "avatar_url": "https://avatars0.githubusercontent.com/u/189618?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/greboid",
      "html_url": "https://github.com/greboid",
      "followers_url": "https://api.github.com/users/greboid/followers",
      "following_url": "https://api.github.com/users/greboid/following{/other_user}",
      "gists_url": "https://api.github.com/users/greboid/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/greboid/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/greboid/subscriptions",
      "organizations_url": "https://api.github.com/users/greboid/orgs",
      "repos_url": "https://api.github.com/users/greboid/repos",
      "events_url": "https://api.github.com/users/greboid/events{/privacy}",
      "received_events_url": "https://api.github.com/users/greboid/received_events",
      "type": "User",
      "site_admin": false
    },
    "html_url": "https://github.com/greboid/irc",
    "description": "Plugin based IRC library for Go",
    "fork": false,
    "url": "https://api.github.com/repos/greboid/irc",
    "forks_url": "https://api.github.com/repos/greboid/irc/forks",
    "keys_url": "https://api.github.com/repos/greboid/irc/keys{/key_id}",
    "collaborators_url": "https://api.github.com/repos/greboid/irc/collaborators{/collaborator}",
    "teams_url": "https://api.github.com/repos/greboid/irc/teams",
    "hooks_url": "https://api.github.com/repos/greboid/irc/hooks",
    "issue_events_url": "https://api.github.com/repos/greboid/irc/issues/events{/number}",
    "events_url": "https://api.github.com/repos/greboid/irc/events",
    "assignees_url": "https://api.github.com/repos/greboid/irc/assignees{/user}",
    "branches_url": "https://api.github.com/repos/greboid/irc/branches{/branch}",
    "tags_url": "https://api.github.com/repos/greboid/irc/tags",
    "blobs_url": "https://api.github.com/repos/greboid/irc/git/blobs{/sha}",
    "git_tags_url": "https://api.github.com/repos/greboid/irc/git/tags{/sha}",
    "git_refs_url": "https://api.github.com/repos/greboid/irc/git/refs{/sha}",
    "trees_url": "https://api.github.com/repos/greboid/irc/git/trees{/sha}",
    "statuses_url": "https://api.github.com/repos/greboid/irc/statuses/{sha}",
    "languages_url": "https://api.github.com/repos/greboid/irc/languages",
    "stargazers_url": "https://api.github.com/repos/greboid/irc/stargazers",
    "contributors_url": "https://api.github.com/repos/greboid/irc/contributors",
    "subscribers_url": "https://api.github.com/repos/greboid/irc/subscribers",
    "subscription_url": "https://api.github.com/repos/greboid/irc/subscription",
    "commits_url": "https://api.github.com/repos/greboid/irc/commits{/sha}",
    "git_commits_url": "https://api.github.com/repos/greboid/irc/git/commits{/sha}",
    "comments_url": "https://api.github.com/repos/greboid/irc/comments{/number}",
    "issue_comment_url": "https://api.github.com/repos/greboid/irc/issues/comments{/number}",
    "contents_url": "https://api.github.com/repos/greboid/irc/contents/{+path}",
    "compare_url": "https://api.github.com/repos/greboid/irc/compare/{base}...{head}",
    "merges_url": "https://api.github.com/repos/greboid/irc/merges",
    "archive_url": "https://api.github.com/repos/greboid/irc/{archive_format}{/ref}",
    "downloads_url": "https://api.github.com/repos/greboid/irc/downloads",
    "issues_url": "https://api.github.com/repos/greboid/irc/issues{/number}",
    "pulls_url": "https://api.github.com/repos/greboid/irc/pulls{/number}",
    "milestones_url": "https://api.github.com/repos/greboid/irc/milestones{/number}",
    "notifications_url": "https://api.github.com/repos/greboid/irc/notifications{?since,all,participating}",
    "labels_url": "https://api.github.com/repos/greboid/irc/labels{/name}",
    "releases_url": "https://api.github.com/repos/greboid/irc/releases{/id}",
    "deployments_url": "https://api.github.com/repos/greboid/irc/deployments",
    "created_at": "2020-02-26T19:09:37Z",
    "updated_at": "2020-03-30T22:37:10Z",
    "pushed_at": "2020-03-31T10:23:17Z",
    "git_url": "git://github.com/greboid/irc.git",
    "ssh_url": "git@github.com:greboid/irc.git",
    "clone_url": "https://github.com/greboid/irc.git",
    "svn_url": "https://github.com/greboid/irc",
    "homepage": "",
    "size": 228,
    "stargazers_count": 0,
    "watchers_count": 0,
    "language": "Go",
    "has_issues": true,
    "has_projects": true,
    "has_downloads": true,
    "has_wiki": true,
    "has_pages": false,
    "forks_count": 1,
    "mirror_url": null,
    "archived": false,
    "disabled": false,
    "open_issues_count": 1,
    "license": {
      "key": "mit",
      "name": "MIT License",
      "spdx_id": "MIT",
      "url": "https://api.github.com/licenses/mit",
      "node_id": "MDc6TGljZW5zZTEz"
    },
    "forks": 1,
    "open_issues": 1,
    "watchers": 0,
    "default_branch": "master"
  },
  "sender": {
    "login": "greboid",
    "id": 189618,
    "node_id": "MDQ6VXNlcjE4OTYxOA==",
    "avatar_url": "https://avatars0.githubusercontent.com/u/189618?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/greboid",
    "html_url": "https://github.com/greboid",
    "followers_url": "https://api.github.com/users/greboid/followers",
    "following_url": "https://api.github.com/users/greboid/following{/other_user}",
    "gists_url": "https://api.github.com/users/greboid/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/greboid/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/greboid/subscriptions",
    "organizations_url": "https://api.github.com/users/greboid/orgs",
    "repos_url": "https://api.github.com/users/greboid/repos",
    "events_url": "https://api.github.com/users/greboid/events{/privacy}",
    "received_events_url": "https://api.github.com/users/greboid/received_events",
    "type": "User",
    "site_admin": false
  }
}