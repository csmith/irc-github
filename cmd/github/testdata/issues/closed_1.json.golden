[greboid/irc] greboid closed issue [BUG] Panic when using streaming API - https://github.com/greboid/irc/issues/9